flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServicesTest.java

```diff
@@ -235,8 +235,7 @@ private void runCleanupTestWithJob(
             final LeaderElectionService jobManagerLeaderElectionService =
                     zooKeeperHaServices.getJobManagerLeaderElectionService(jobId);
 
-            final RunningJobsRegistry runningJobsRegistry =
-                    zooKeeperHaServices.getRunningJobsRegistry();
+            final JobResultStore jobResultStore = zooKeeperHaServices.getJobResultStore();
```

Comment:
unused, should we also incorporate dirty -> clean into the lifecycle?

Reply:
good catch. We should remove it. The test is about cleaning up the znodes. It's just about creating some paths. The integration of this component into the test becomes obsolete with the `JobResultStore` replacing the `RunningJobRegistry` since no ZK-related artifacts are created by the `JobResultStore`.